components/raftstore/src/store/fsm/peer.rs

```diff
@@ -3520,11 +3509,19 @@ where
             return;
         }
 
+        self.register_split_region_check_tick();
```

Comment:
seems it should be ahead of the above check?

Reply:
No. If we do not pass the check 
``` if self.fsm.peer.has_calculated_region_size
            && self.fsm.peer.compaction_declined_bytes < self.ctx.cfg.region_split_check_diff.0
            && self.fsm.peer.size_diff_hint < self.ctx.cfg.region_split_check_diff.0```, TiKV will never check again util more entries are appended.

Reply:
What if more entries are appended but committed or applied after next split check?